api/core/entities/provider_configuration.py

```diff
@@ -1807,26 +1807,30 @@ def to_list(self) -> list[ProviderConfiguration]:
         """
         return list(self.values())
 
-    def __getitem__(self, key):
+    def __getitem__(self, key) -> ProviderConfiguration:
         if "/" not in key:
             key = str(ModelProviderID(key))
 
         return self.configurations[key]
 
-    def __setitem__(self, key, value):
+    def __setitem__(self, key: str, value: ProviderConfiguration) -> None:
         self.configurations[key] = value
 
-    def __iter__(self):
+    def __iter__(self) -> Iterator[str]:
```

Comment:
Return type "Iterator[str]" of "__iter__" incompatible with return type "Generator[tuple[str, Any], None, None]" in supertype "pydantic.main.BaseModel"